pythonx/jedi_vim.py

```diff
@@ -158,36 +158,39 @@ def wrapper(*args, **kwargs):
     return func_receiver
 
 
-last_force_python_error = None
+current_environment = (None, None)
 
 
-def get_environment():
-    global last_force_python_error
+def get_environment(use_cache=True):
+    global current_environment
+
+    vim_force_python_version = vim_eval("g:jedi#force_py_version")
+    if use_cache and vim_force_python_version == current_environment[0]:
+        return current_environment[1]
 
-    force_python_version = vim_eval("g:jedi#force_py_version")
     environment = None
-    if force_python_version != "auto":
+    if vim_force_python_version == "auto":
+        environment = jedi.api.environment.get_cached_default_environment()
```

Comment:
If we cache anyway, we can just use jedi.get_default_environment() instead. It's the official API, this one is kind of hidden and not really public.